server/game/cards/13-O4B/Fort51.js

```diff
@@ -0,0 +1,67 @@
+const GameActions = require('../../GameActions/index.js');
+const OutfitCard = require('../../outfitcard.js');
+
+class Fort51 extends OutfitCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                onGadgetInvented: event => event.gadget.hasKeyword('Weapon') || event.gadget.getType() === 'dude' && event.player === this.controller
```

Comment:
The `and` (&&) operator has a priority so this condition would not check the controller in case the gadget has 'weapon' keyword, and will trigger also if opponent invents weapon gadget.
```suggestion
                onGadgetInvented: event => (event.gadget.hasKeyword('Weapon') || event.gadget.getType() === 'dude') && event.player === this.controller
```